src/core/Board.js

```diff
@@ -9,52 +9,65 @@ import styles from "./Board.module.scss";
 export const Board = () => {
   const [shuffledCards, setShuffledCards] = useState(cardFaces());
   const [inputTracker, setInputTracker] = useState([]);
+  const [turnCounter, setTurnCounter] = useState(0);
+  const [matchCounter, setMatchCounter] = useState(0);
+
+  const compareCards = compareArray => {
+    const firstCard = shuffledCards[compareArray[0]].name;
+    const secondCard = shuffledCards[compareArray[1]].name;
+    console.log(firstCard, secondCard);
+
+    if (firstCard === secondCard) {
+      setMatchCounter(matchCounter + 1);
+    } else {
+      setTimeout(() => {
+        const newUpdatedCards = [...shuffledCards];
+        compareArray.forEach(cardIndex => {
+          newUpdatedCards[cardIndex] = flipCard(
+            newUpdatedCards[cardIndex],
+            false
+          );
+        });
+        setShuffledCards(newUpdatedCards);
+      }, 1000);
+    }
+  };
 
   const handleFlip = (key, card) => {
     if (!card.isFaceUp) {
       const updatedCards = [...shuffledCards];
+      const updatedTracker = [...inputTracker];
```

Comment:
Nice variable names! They make your code easy to understand.

Reply:
Thank you :)